### PR TITLE
feat: Reflect blockage of deletion in module status

### PR DESCRIPTION
**Pull request by shorim**
## Description

Changes proposed in this pull request:

- When there is a request to delete the Telemetry CR while orphaned custom resources still exist in the cluster, the Telemetry CR will not be deleted as long as the custom resource still exist and Telemetry `status` will change to the following:
  - `status.state` will be `Warning`, since this is the state which requires an action from the user.
  -  If the oprhaned resource is `MetricPipeline`, the following `status.condition` will appear:
      ```
      lastTransitionTime: "2023-09-15T12:56:39Z"
      message: One or more MetricPipelines still exist
      observedGeneration: 3
      reason: MetricResourceBlocksDeletion
      status: "False"
      type: MetricComponentsHealthy
      ```
  -  If the oprhaned resource is `TracePipeline`, the following `status.condition` will appear:
      ```
      lastTransitionTime: "2023-09-15T12:56:39Z"
      message: One or more TracePipelines still exist
      observedGeneration: 3
      reason: TraceResourceBlocksDeletion
      status: "False"
      type: TraceComponentsHealthy
      ```
  -  If the oprhaned resource is `LogPipeline` or `LogParsers`, the following `status.condition` will appear:
      ```
      lastTransitionTime: "2023-09-15T12:56:39Z"
      message: One or more LogPipelines/LogParsers still exist
      observedGeneration: 3
      reason: LogResourceBlocksDeletion
      status: "False"
      type: LogComponentsHealthy
      ```
   
- Updated the documentation for Telemetry status conditions
- Modified existing unit and e2e tests and added new unit and e2e tests.

## Related Issues and Documents

Closes:

Related issues:
#381
## Traceability

- [x] The PR is linked to a GitHub Issue.
- [x] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [x] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [x] I have augmented the test suite that proves my fix is effective or that my feature works.
- [x] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: internal/reconciler/status.go [hunk 1]
```diff
@@ -5,29 +5,35 @@ const (
 	ReasonReferencedSecretMissing = "ReferencedSecretMissing"
 	ReasonWaitingForLock          = "WaitingForLock"
 
-	ReasonFluentBitDSNotReady = "FluentBitDaemonSetNotReady"
-	ReasonFluentBitDSReady    = "FluentBitDaemonSetReady"
+	ReasonFluentBitDSNotReady          = "FluentBitDaemonSetNotReady"
+	ReasonFluentBitDSReady             = "FluentBitDaemonSetReady"
+	ReasonLogComponentsDeletionBlocked = "LogComponentsDeletionBlocked"
```
Comment (chrkl) [inline internal/reconciler/status.go:10]: I think it's not obvious from this naming what is blocking the deletion of what. Maybe we can use something like `LogPipelineBlocksDeletion`. Please think about better ideas.

File: internal/reconciler/telemetry/reconciler.go [hunk 2]
```diff
@@ -92,8 +92,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile webhook: %w", err)
 	}
 
-	// Keep requeueing until dependent resources are removed
-	requeue := telemetry.Status.State == operatorv1alpha1.StateError
+	// Keep requeuing as long as the Telemetry CR is in a "Warning" state
```
[the remark this answers is not in the document]
Comment (shorim) [reply internal/reconciler/telemetry/reconciler.go]: As discussed, there are other situations as well that will make the Telemetry CR in a "Warning" state. But the comment is still obvious from the code, so we decided to just remove it.

**Review by chrkl — DISMISSED** [review]

File: docs/user/resources/01-telemetry.md [hunk 3]
```diff
@@ -78,6 +78,6 @@ For details, see the [Telemetry specification file](https://github.com/kyma-proj
 | **endpoints.&#x200b;traces**  | object |  |
 | **endpoints.&#x200b;traces.&#x200b;grpc**  | string |  |
 | **endpoints.&#x200b;traces.&#x200b;http**  | string |  |
-| **state** (required) | string | State signifies current state of Module CR. Value can be one of ("Ready", "Processing", "Error", "Deleting", "Warning"). |
+| **state** (required) | string | State signifies current state of Module CR. Value can be one of ("Ready", "Deleting", "Warning"). |
```
Comment (IwonaLanger) [inline docs/user/resources/01-telemetry.md:81]:
```suggestion
| **state** (required) | string | State signifies current state of Module CR. Value can be one of these three: `Ready`, `Deleting`, or `Warning`. |
```

**Review by chrkl — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**